### ISSUE TITLE
Training interruption and accuracy issues

### ISSUE THREAD

**Issue by xyandtyh** [YkiWu/EmbodiedOcc#12]
Question 1:
We use this commandtorchrun --nproc_per_node=8 train_embodied.py --py-config config/train_embodied_config.pyto start training, and then we find that the training is interrupted after about 2 epochs, and after the training is interrupted, there is a problem that the GPU resources are not released. The processes list shows no related processes, but the GPU information shows that there is video memory occupation. Have you encountered similar problems?

<img width="815" height="858" alt="Image" src="https://github.com/user-attachments/assets/1afccffb-6e68-497b-9c4c-33d3947929b7" />

Question 2:
We launched the training using the same configuration as yours with the command torchrun --nproc_per_node=8 train_embodied.py --py-config config/train_embodied_config.py, but the mIoU value differs significantly from what was reported in your paper.

<img width="1511" height="793" alt="Image" src="https://github.com/user-attachments/assets/7e5b4b5d-59ab-4398-95bd-bbd52d8047f2" />

I'm very sorry to bother you and sincerely look forward to your reply!

**Comment by YkiWu** [comment]
> Question 1: We use this commandtorchrun --nproc_per_node=8 train_embodied.py --py-config config/train_embodied_config.pyto start training, and then we find that the training is interrupted after about 2 epochs, and after the training is interrupted, there is a problem that the GPU resources are not released. The processes list shows no related processes, but the GPU information shows that there is video memory occupation. Have you encountered similar problems?
> 
> <img alt="Image" width="815" height="858" src="https://private-user-images.githubusercontent.com/113224688/519774616-1afccffb-6e68-497b-9c4c-33d3947929b7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQyODU4NzAsIm5iZiI6MTc2NDI4NTU3MCwicGF0aCI6Ii8xMTMyMjQ2ODgvNTE5Nzc0NjE2LTFhZmNjZmZiLTZlNjgtNDk3Yi05YzRjLTMzZDM5NDc5MjliNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyN1QyMzE5MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNmYyMTI2YTQxYWRlMzdkMWVkNGNhNWQxMDJjYWZmZGZjYzM5OTlhMTZmZThhYTVhYTM2OTJjYTAxMTE4NzNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HRX_qvl5i8eDfzIcYhjajQZq_1fsHsJXRET164gz65s">
> Question 2: We launched the training using the same configuration as yours with the command torchrun --nproc_per_node=8 train_embodied.py --py-config config/train_embodied_config.py, but the mIoU value differs significantly from what was reported in your paper.
> 
> <img alt="Image" width="1511" height="793" src="https://private-user-images.githubusercontent.com/113224688/519774731-7e5b4b5d-59ab-4398-95bd-bbd52d8047f2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQyODU4NzAsIm5iZiI6MTc2NDI4NTU3MCwicGF0aCI6Ii8xMTMyMjQ2ODgvNTE5Nzc0NzMxLTdlNWI0YjVkLTU5YWItNDM5OC05NWJkLWJiZDUyZDgwNDdmMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyN1QyMzE5MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNmRjZTYyZjU4MDE1M2E4MWQxMWQ5ZmE3MjIxMTljM2Y4YzEyOWY5NDNhZTZiNzY4MDRjNjRkNjY2Yzg2N2Q0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Rut2DtagTAANhS115bTULWqODpJG93RUcWKmv4tdZyQ">
> I'm very sorry to bother you and sincerely look forward to your reply!

Re Q1: I haven't encountered this issue myself, sorry.
Re Q2: If the monocular checkpoint is loaded correctly, this issue likely stems from running the test on 8 GPUs. Because the test set has only 16 scenes, an 8-GPU setup results in the main process recording only 2 scenes, leading to zero metrics for many categories. Please try running the evaluation on a single GPU.

**Comment by xyandtyh** [comment]
<img width="1539" height="719" alt="Image" src="https://github.com/user-attachments/assets/25e1358b-12ad-4d9e-a591-e735139ec48f" /> Hello, I'm really sorry to bother you again. I followed the configuration steps on GitHub to set up the environment and code to match your settings, then trained on 8 GPUs as you suggested, and tested on 1 GPU during the testing phase. The final mIoU was 13%, which is very troubling for me. However, when I used the command `torchrun --nproc_per_node=8 train_mono.py --py-config config/train_mono_config.py`, the test results under monocular view matched yours. I suspect that when using the command `torchrun --nproc_per_node=8 train_embodied.py --py-config config/train_embodied_config.py` for training, I might have missed some additional steps, leading to certain pre-trained models not being loaded. The pre-trained models I've downloaded include: (EmbodiedOcc/checkpoints/finetune_scannet_depthanythingv2.pth, rwightman_gen-efficientnet-pytorch_master/tf_efficientnet_b7_ns-1dbc32de.pth). Could there be some configuration steps missing in the documentation? I apologize for the trouble and look forward to your reply.

**Comment by YkiWu** [comment]
> <img alt="Image" width="1539" height="719" src="https://private-user-images.githubusercontent.com/113224688/520394178-25e1358b-12ad-4d9e-a591-e735139ec48f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ1NDYzMTMsIm5iZiI6MTc2NDU0NjAxMywicGF0aCI6Ii8xMTMyMjQ2ODgvNTIwMzk0MTc4LTI1ZTEzNThiLTEyYWQtNGQ5ZS1hNTkxLWU3MzUxMzllYzQ4Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEzMFQyMzQwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMDM3NmUyNmNjMDNiYzczYzNmOWIxMTQ5MzM4YmYwOTUyZGZmYzlhZGU0MTAwN2YxMzZhM2RmYTgwY2FiODMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nIXZyPJGJvrDso0UwbswuJ4qQ6I9Ez0J31G47fGac38"> Hello, I'm really sorry to bother you again. I followed the configuration steps on GitHub to set up the environment and code to match your settings, then trained on 8 GPUs as you suggested, and tested on 1 GPU during the testing phase. The final mIoU was 13%, which is very troubling for me. However, when I used the command `torchrun --nproc_per_node=8 train_mono.py --py-config config/train_mono_config.py`, the test results under monocular view matched yours. I suspect that when using the command `torchrun --nproc_per_node=8 train_embodied.py --py-config config/train_embodied_config.py` for training, I might have missed some additional steps, leading to certain pre-trained models not being loaded. The pre-trained models I've downloaded include: (EmbodiedOcc/checkpoints/finetune_scannet_depthanythingv2.pth, rwightman_gen-efficientnet-pytorch_master/tf_efficientnet_b7_ns-1dbc32de.pth). Could there be some configuration steps missing in the documentation? I apologize for the trouble and look forward to your reply.

Please make sure that you have loaded the mono checkpoint when training the embodied version. And please use 1 GPU for evaluation.
